consensus/reactor.go

```diff
@@ -214,7 +218,10 @@ func (conR *ConsensusReactor) Receive(chID byte, src p2p.Peer, msgBytes []byte)
 	conR.Logger.Debug("Receive", "src", src, "chId", chID, "msg", msg)
 
 	// Get peer states
-	ps := src.Get(types.PeerStateKey).(*PeerState)
+	ps, ok := src.Get(types.PeerStateKey).(*PeerState)
+	if !ok {
+		panic(fmt.Sprintf("Peer %v has no state", src))
```

Comment:
?? we only panic if src has no state (`!ok`). I don't understand what do you mean by "anyways"

Reply:
My understanding is we only get `!ok` if `ps := src.Get(types.PeerStateKey).(*PeerState)` would have paniced. The point of the `ok` is to prevent the native panic and do something else. But here we're just explicilty panicing - so why even bother with the `ok`?

Reply:
ah, right. probably not much sense except better error message